src/uuid.ts

```diff
@@ -0,0 +1,57 @@
+const UUID_RX = /^(?:[0-9a-f]{8}-[0-9a-f]{4}-[1-5][0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}|00000000-0000-0000-0000-000000000000)$/i;
+
+/** @internal */
+function formatUUID(uuid: string): string {
+  uuid = uuid.replace(/^.+:/, '');
```

Comment:
I think this is unnecessary
This is funky wording from the spec:
```
Parsers MUST accept textual representations of UUIDs that omit the URN prefix (usually urn:uuid:)
```
"MUST accept UUIDs that **_omit_** the prefix": It's like a positive/negative but I read it as the parser should not accept uuids with such a prefix.

Reply:
Originally I was trying to match python.

```python
>>> import uuid, base64
>>> print(base64.b64encode(uuid.UUID('c8edabc3-f738-4ca3-b68d-ab92a91478a3').bytes))
yO2rw/c4TKO2jauSqRR4ow==
>>> print(base64.b64encode(uuid.UUID('urn:uuid:c8edabc3-f738-4ca3-b68d-ab92a91478a3').bytes))
yO2rw/c4TKO2jauSqRR4ow==
>>> print(base64.b64encode(uuid.UUID('c8edabc3f7384ca3b68dab92a91478a3').bytes))
yO2rw/c4TKO2jauSqRR4ow==
```

The requirements are:

```
urn:uuid:9b1deb4d-3b7d-4bad-9bdd-2b0d7b3dcb6d MAY return mx3rTTt9S62b3SsNez3LbQ==
9b1deb4d-3b7d-4bad-9bdd-2b0d7b3dcb6d MUST return mx3rTTt9S62b3SsNez3LbQ==
9b1deb4d3b7d4bad9bdd2b0d7b3dcb6d MAY return mx3rTTt9S62b3SsNez3LbQ==
```

It was my decision to play it safe by supporting this, I will remove the `format` function I didn't think it would raise that may eyebrows. Now it just seems like extra work, that is optional.

Reply:
Yea my initial reaction would've been the same. I guess when dealing with computer data formats (even though json is mostly readable) erring on the side of stricter feels more correct for the circumstance.